activerecord/test/cases/query_comment_test.rb

```diff
@@ -0,0 +1,177 @@
+    assert_sql(%r{/\*line:#{__FILE__}:[0-9]+:in `block in test_last_line_component'\*/$}) do
+      Dashboard.first
+    end
+  ensure
+    comment_context.components = components
+  end
+
+  def test_default_lines_to_ignore_regex
+    line = "/gems/a_gem/lib/a_gem.rb:1:in `some_method'"
+    call_stack = [line] + caller
+
+    assert_match(
+      call_stack.detect { |line| line !~ comment_context::DEFAULT_LINES_TO_IGNORE_REGEX },
+      line
+    )
+  end
+
+  def test_last_line_component_with_lines_to_ignore
+    comment_context.lines_to_ignore = /foo bar/
+    comment_context.components = [:line]
+    # Because "lines_to_ignore" does not include "active_support" any more, the extracted
+    # line will be from core_ext#try
+    assert_sql(%r{/\*line:.*/active_support/core_ext/object/try.rb:[0-9]+}) do
+      Dashboard.first
+    end
+  ensure
+    comment_context.lines_to_ignore = nil
+  end
+
+  def test_config_db_host
+    skip if current_adapter?(:SQLite3Adapter)
+
+    # mock the connection config to expose a socket value
+    db_config = ActiveRecord::Base.connection.pool.db_config
+    db_config.stub(:configuration_hash, {:host => "localhost"}) do
+      comment_context.components = [:db_host]
+      assert_sql(%r{/\*db_host:localhost}) do
+        Dashboard.first
+      end
+    end
+  end
+
+  def test_config_database
+    comment_context.components = [:database]
+    assert_sql(%r{/\*database:.*(fixture_database|activerecord_unittest)+}) do
+      Dashboard.first
+    end
+  end
+
+  def test_config_socket
+    skip unless current_adapter?(:Mysql2Adapter)
+    # mock the connection config to expose a socket value
+    db_config = ActiveRecord::Base.connection.pool.db_config
+    db_config.stub(:configuration_hash, {:socket => "comment_test_socket"}) do
+      comment_context.components = [:socket]
+      assert_sql(%r{/\*socket:comment_test_socket}) do
+        Dashboard.first
+      end
+    end
+  end
+
+  def test_retrieves_comment_from_cache_when_enabled_and_set
+    comment_context.cache_query_comment = true
+    comment_context.components = [:application]
+    assert_equal "/*application:rails*/", comment_context.comment
+    comment_context.stub(:cached_comment, "/*cached_comment*/") do
+      assert_equal "/*cached_comment*/", comment_context.comment
+    end
+  ensure
+    comment_context.clear_comment_cache!
+    comment_context.cache_query_comment = false
+  end
+
+  def test_resets_cache_on_context_update
+    comment_context.cache_query_comment = true
+    comment_context.components = [:application]
+    assert_equal "/*application:rails*/", comment_context.comment
+    comment_context.set(application_name: "new_name")
+    assert_nil comment_context.cached_comment
+    assert_equal "/*application:new_name*/", comment_context.comment
+  ensure
+    comment_context.clear_comment_cache!
+    comment_context.cache_query_comment = false
+    comment_context.set(application_name: nil)
+  end
```

Comment:
This wouldn't get rejected but as a style thing I like to separate test setup from assertions. I'd rewrite this like:

```ruby
  def test_resets_cache_on_context_update
    comment_context.cache_query_comment = true
    comment_context.components = [:application]

    assert_equal "/*application:rails*/", comment_context.comment

    comment_context.set(application_name: "new_name")

    assert_nil comment_context.cached_comment
    assert_equal "/*application:new_name*/", comment_context.comment
  ensure
    comment_context.clear_comment_cache!
    comment_context.cache_query_comment = false
    comment_context.set(application_name: nil)
  end
```